### PR TITLE
606 Correct links to prefiltered listings

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Addresses feedback on [606](https://projects.torchbox.com/projects/rca-website-rebuild/tickets/606#update-65991341)

### CODE REVIEW

File: rca/schools/models.py [hunk 1]
```diff
@@ -509,9 +510,23 @@ def get_related_programmes(self):
 
     def get_programme_index_link(self):
         ProgrammeIndexPage = apps.get_model("programmes", "ProgrammeIndexPage")
-        programm_index = ProgrammeIndexPage.objects.live().first()
-        if programm_index:
-            return f"{programm_index.get_url()}?category=related_schools_and_research_pages&value={self.pk}-{self.slug}"
+        programme_index = ProgrammeIndexPage.objects.live().first()
```
Comment (richbrennan) [inline rca/schools/models.py:513]: If this returns `None` the next line will error, does this need to be checked?

File: rca/schools/models.py [hunk 2]
```diff
@@ -509,9 +510,23 @@ def get_related_programmes(self):
...
+        """
+        ProgrammeType = apps.get_model("programmes", "ProgrammeType")
+        short_course_type = ProgrammeType.objects.filter(
+            display_name="Short course"
+        ).first()
+        if not short_course_type:
+            return
+        return (
+            f"{self.get_programme_index_link()}?category=programme_type&"
+            f"value={str(short_course_type.id)}-{slugify(short_course_type.display_name)}"
```
Comment (richbrennan) [inline rca/schools/models.py:528]: Does `str()` need to be called here?

File: rca/schools/models.py [hunk 3]
```diff
@@ -509,9 +510,23 @@ def get_related_programmes(self):
...
+
+    def get_short_courses_index_link(self):
+        """Returns a link to the programme index page filtered by the
+        Short Course ProgrammeType
+        """
+        ProgrammeType = apps.get_model("programmes", "ProgrammeType")
+        short_course_type = ProgrammeType.objects.filter(
+            display_name="Short course"
+        ).first()
+        if not short_course_type:
```
Comment (richbrennan) [inline rca/schools/models.py:524]: How about

```
if short_course_type:
        return (
            f"{self.get_programme_index_link()}?category=programme_type&"
            f"value={str(short_course_type.id)}-{slugify(short_course_type.display_name)}"
        )
```

**Review by richbrennan — COMMENTED** [review]
A few comments, thanks Kevin!

File: rca/schools/models.py (hunk 2 above)
Comment (kevinhowbrook) [reply rca/schools/models.py:528]: Nope - hangover from when I didn't have this as an `f` string

**Pull request merged**